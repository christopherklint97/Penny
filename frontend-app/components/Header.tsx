import styles from './Header.module.css'
import { Navbar, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap'
import { BACKEND_API } from '../config'
import { useSelector } from 'react-redux'
import { RootState } from '../redux/index'

export function Header() {
  const isLoggedIn = useSelector((state: RootState) => state.login.loggedIn)

  return (
    <Navbar expand="md" className={`${styles.blueBg} + sticky-top`}>
      <NavbarBrand href="/" className={`${styles.whiteText} yellowtail`}>
        Penny
      </NavbarBrand>
      <Nav className="ml-auto" navbar>
        <NavItem>
          {!isLoggedIn ? (
            <NavLink
              href={`${BACKEND_API}/auth/facebook`}
              className={styles.whiteText}
            >
              Login
            </NavLink>
          ) : (
            <NavLink href={`api/logout`} className={styles.whiteText}>
              Logout
            </NavLink>
          )}
        </NavItem>
      </Nav>
    </Navbar>
  )
}
