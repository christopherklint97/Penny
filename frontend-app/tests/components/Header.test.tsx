import React from 'react'
import { render } from '../testUtils'
import Header from '../../components/Header'

describe('Header component', () => {
  it('renders without crashing', function () {
    render(<Header />)
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<Header />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
