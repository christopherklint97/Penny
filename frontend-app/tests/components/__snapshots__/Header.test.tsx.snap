// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="h-14 p-3 bg-blue-300 text-gray-50"
  >
    <div
      class="flex flex-row justify-between md:w-5/6 mx-auto"
    >
      <a
        class="my-auto"
        href="/"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAiIGhlaWdodD0iMzAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block;"
            />
          </div>
          <img
            alt=""
            class="rounded-md"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </a>
      <button
        class="bg-gray-100 hover:bg-gray-300 text-gray-900 px-3 py-1 rounded-md"
      >
        Login
      </button>
    </div>
  </nav>
</DocumentFragment>
`;
